alns/AdaptiveStrategy.py

```diff
@@ -0,0 +1,31 @@
+class AdaptiveStrategy:
```

Comment:
It's still a bit unclear how the _strategy_ part of this should be shaped. I'm inclined to allow this to be a passed-in function, accepting the current weight and the update weight such that a new computation can be achieved. Maybe one or two defaults are sensible as well (e.g. `convex_combination` which is currently used, and a simple addition rule which adds the update weight to the current weight).

Reply:
I should think about this a bit longer.